ample/modelling/rosetta_model.py

```diff
@@ -653,7 +786,7 @@ def process_cmd_list(cmds):
         cmd_str = ""
         for i, c in enumerate(cmds):
             if c.startswith('-') and i > 0:
-                 cmd_str += os.linesep
+                cmd_str += os.linesep
```

Comment:
bad bad bad idea. this will create a new string object with every iteration.

Reply:
I"m not sure I understand - can you explain further?

Reply:
python string concatenation creates a new string object every time you concatenate a new string. as such, you're creating `N` string objects, where `N = len(cmds) + 1`.  you should use the original `list` to modify each element as necessary, then do a single `'  '.join(cmds)` or similar. Even creating a new array and appending is more efficient.

```python
>>> a = 'test'
>>> print(id(a))
4316062360
>>> a += 'foo'
>>> a
'testfoo'
>>> print(id(a))
4316062416
>>> a += 'bar'
>>> a
'testfoobar'
>>> print(id(a))
4314229168
```